src/librustc_typeck/astconv/mod.rs

```diff
@@ -138,8 +134,10 @@ pub enum ExplicitLateBound {
     No,
 }
 
+/// Denotes the "position" of a generic argument, indicating if it is a generic type,
+/// generic function or generic method call.
 #[derive(Copy, Clone, PartialEq)]
-enum GenericArgPosition {
+pub enum GenericArgPosition {
```

Comment:
is this really public? If it's not used in this file anymore, move it to the file where it is used, otherwise maybe use `pub(crate)`

Reply:
It is used in that file. I'll use `pub(crate)`

Reply:
Oh it can't be anything else than public. Some other functions that had to become public use that enum as an argument.

Reply:
These functions can also become `pub(crate)`. The reason why I'd prefer them to be `pub(crate)` (or even `pub(super)`) is that someone reading the code doesn't have to wonder whether these functions are called from arbitrary places in the compiler.